File: source/dotnet5-petclinic-rest/PetClinic.Application/Dtos/PetDTO.cs
using System;
using System.Collections.Generic;
using AutoMapper;
using Intent.RoslynWeaver.Attributes;
using PetClinic.Application.Common.Mappings;
using PetClinic.Domain.Entities;

[assembly: DefaultIntentManaged(Mode.Fully)]
[assembly: IntentTemplate("Intent.Application.Dtos.DtoModel", Version = "1.0")]

namespace PetClinic.Application.Dtos
{

    public class PetDTO : IMapFrom<Pet>
    {
        public PetDTO()
        {
        }

        public static PetDTO Create(
            int id,
            string name,
            DateTime birthDate,
            int petTypeId,
            string petTypeName,
            int ownerId,
            string ownerFirstName,
            string ownerLastName,
            List<PetVisitDTO> visits)
        {
            return new PetDTO
            {
                Id = id,
                Name = name,
                BirthDate = birthDate,
                PetTypeId = petTypeId,
                PetTypeName = petTypeName,
                OwnerId = ownerId,
                OwnerFirstName = ownerFirstName,
                OwnerLastName = ownerLastName,
                Visits = visits,
            };
        }

        public int Id { get; set; }

        public string Name { get; set; }

        public DateTime BirthDate { get; set; }

        public int PetTypeId { get; set; }

        public string PetTypeName { get; set; }

        public int OwnerId { get; set; }

        public string OwnerFirstName { get; set; }

        public string OwnerLastName { get; set; }

        public List<PetVisitDTO> Visits { get; set; }

        public void Mapping(Profile profile)
        {
            profile.CreateMap<Pet, PetDTO>()
                .ForMember(d => d.PetTypeId, opt => opt.MapFrom(src => src.PetType.Id))
                .ForMember(d => d.PetTypeName, opt => opt.MapFrom(src => src.PetType.Name))
                .ForMember(d => d.OwnerId, opt => opt.MapFrom(src => src.Owner.Id))
                .ForMember(d => d.OwnerFirstName, opt => opt.MapFrom(src => src.Owner.FirstName))
                .ForMember(d => d.OwnerLastName, opt => opt.MapFrom(src => src.Owner.LastName));
        }
    }
}

File: source/nestjs-petclinic-rest/src/services/dto/pet.dto.ts
import { PetVisitDTO } from './pet-visit.dto';
import { Pet } from './../../domain/entities/pet.entity';
import { ApiProperty } from '@nestjs/swagger';
import { Type } from 'class-transformer';

export class PetDTO
{
  @ApiProperty()
  id: number;

  @ApiProperty()
  name: string;

  @ApiProperty()
  @Type(() => Date)
  birthDate: Date;

  @ApiProperty()
  petTypeId: number;

  @ApiProperty()
  petTypeName: string;

  @ApiProperty()
  ownerId: number;

  @ApiProperty()
  ownerFirstName: string;

  @ApiProperty()
  ownerLastName: string;

  @ApiProperty()
  visits: PetVisitDTO[];

  static fromPet(pet: Pet) {
    if (pet == null) {
        return null;
    }
    const dto = new PetDTO();
    dto.id = pet.id;
    dto.name = pet.name;
    dto.birthDate = pet.birthDate;
    dto.petTypeId = pet.petType?.id;
    dto.petTypeName = pet.petType?.name;
    dto.ownerId = pet.owner?.id;
    dto.ownerFirstName = pet.owner?.firstName;
    dto.ownerLastName = pet.owner?.lastName;
    dto.visits = pet.visits?.map(x => PetVisitDTO.fromVisit(x));
    return dto;
  }

  static requiredRelations: string[] = ["petType", "owner", "visits"];
}